crates/hir_ty/src/infer.rs

```diff
@@ -407,6 +407,12 @@ impl<'a> InferenceContext<'a> {
         for (_, subst) in result.method_resolutions.values_mut() {
             *subst = self.table.resolve_completely(subst.clone());
         }
+        for adjustment in result.expr_adjustments.values_mut().flatten() {
```

Comment:
We can avoid the clones here and above using `mem::replace`, do you think it's worth it?

Reply:
No, cloning Chalk types is cheap.

Reply:
Yeah, I suppose we'd still have to intern (or at least clonel the error type.